### PR TITLE
Add the winfstest test suite to winfspy

### PR DESCRIPTION
**Pull request by vxgmichel**
Address #11 

### Changes:
- move `memory.py` to `winfspy.memfs`
- improve `winfspy.memfs`
- add the `winfstest` test suite to `winfspy.tests.winfstest`
- adapt packaging and CI accordingly

### Rationale

Winfstest provides black box testing for an arbitrary file system. 

Porting it to python and integrating it into winfspy provides several advantages:
- pytest integration with nice error reports
- it allows us to easily test winfspy by throwing `winfspy.memfs` against it
- by shipping it with winfspy, it is easily usable for other projects:
    ```shell
    $ pip install winfspy[test]
    $ pytest --pyargs winfspy.tests.winfstest --file-system-path X:
    ```

### TODO

Only the first test case is supported at the moment. It is a work in progress.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -70,3 +70,7 @@ deploy_script:
   - set HOME=%USERPROFILE%
   - ps: If ($env:APPVEYOR_REPO_TAG -eq "true") { Invoke-Expression "twine upload --verbose --skip-existing dist/*.whl" } Else { write-output "Not on a tag, won't deploy to pypi"}
   - echo "Finished Artifact Deployment"
+
+on_success:
+  - pip install -U --user codecov
+  - python -m codecov
```
Comment (c4ffein) [inline .appveyor.yml]: The lack of an ending newline triggers my ocd

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+coverage:
+  status:
+    project: off
```
Comment (c4ffein) [inline .codecov.yml:3]: We don't use coverage anymore?

File: src/winfspy/memfs.py [hunk 3]
```diff
@@ -186,12 +192,15 @@ def create(
         except KeyError:
             raise NTStatusObjectNameNotFound()
 
-        # TODO: handle file_attributes
+        # File/Folder already exists
+        if file_name in self._entries:
+            raise NTStatusObjectNameCollision()
 
+        # TODO: handle file_attributes
```
Comment (c4ffein) [inline src/winfspy/memfs.py]: Still a TODO?

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -10,9 +10,7 @@ repos:
     rev: 3.7.9
     hooks:
     - id: flake8
-      args:
-        - "--max-line-length=100"
-        - "--ignore=E203,E266,E501,W503"
+      exclude: .*\.t
```
Comment (c4ffein) [inline .pre-commit-config.yaml:13]: Why exclude .t from linter if they still are python files?

File: src/winfspy/memfs.py (hunk 3 above)
Comment (vxgmichel) [reply src/winfspy/memfs.py]: Good catch!

File: .pre-commit-config.yaml (hunk 4 above)
Comment (vxgmichel) [reply .pre-commit-config.yaml:13]: Because they've been copied from elsewhere and they don't comply with flake8 :)

File: .codecov.yml (hunk 2 above)
Comment (vxgmichel) [reply .codecov.yml:3]: It only disables the CI checks (e.g failing the PR if the coverage decreases)

File: .appveyor.yml (hunk 1 above)
Comment (vxgmichel) [reply .appveyor.yml]: oopsie

**Pull request merged**